app/controllers/articles_controller.rb

```diff
@@ -279,8 +279,8 @@ def handle_notifications(updated)
     if updated && @article.published && @article.saved_changes["published"] == [false, true]
       Notification.send_to_followers(@article, "Published")
     elsif @article.saved_changes["published"] == [true, false]
-      Notification.remove_all_without_delay(notifiable_id: @article.id, notifiable_type: "Article", action: "Published")
-      Notification.remove_all(notifiable_id: @article.id, notifiable_type: "Article", action: "Reaction")
+      Notification.remove_all_by_action_without_delay(notifiable_ids: @article.id, notifiable_type: "Article", action: "Published")
+      Notification.remove_all(notifiable_ids: @article.comments.pluck(:id), notifiable_type: "Comment") if @article.comments.exists?
```

Comment:
Hmm, considering changing `if @article.comments.exists?` to `if Comment.exists?(commentable: article)` or `unless @article.comments_count.zero?`. Thoughts?

The first one is more accurate, but the second one is faster and doesn't have a second query. The cache column might be incorrect though.

Reply:
Sorry, I didn't get why `@article.comments.exists?` errors out, could you elaborate?
I'm also fine with `Comment.exists?(commentable: article)` and have the same concern about the `@article.comments_count.zero?` (though in most cases it will also be fine)

Reply:
Hmm I realized I ran it in regular `pry` and not `rails console`. That combined with the spec failing it didn't work. Let me do a more thorough check; sorry about that.

Reply:
Sigh, turns out I put `@article` which was `nil`, and not `article` which was the correct variable. Thanks for probing 😓